src/agent/src/rpc.rs

```diff
@@ -236,6 +238,30 @@ impl AgentService {
             }
         }
 
+        let linux = oci
+            .linux
+            .as_mut()
+            .ok_or_else(|| anyhow!("Spec didn't contain linux field"))?;
+
+        for specdev in &mut linux.devices {
+            let dev_major_minor = format!("{}:{}", specdev.major, specdev.minor);
+
+            if specdev.path == TRUSTED_STORAGE_DEVICE {
+                let data_integrity = AGENT_CONFIG.data_integrity;
+                info!(
+                    sl(),
+                    "trusted_store device major:min {}, enable data integrity {}",
+                    dev_major_minor,
+                    data_integrity.to_string()
+                );
+
+                Command::new(INIT_TRUSTED_STORAGE)
```

Comment:
Do you think it could make sense to handle this functionality in the CDH? That might tie-in nicely with when we support persistent storage that we need a key to unlock.

Reply:
That's a good idea. Perhaps it would be better to merge this into CDH when we move forward the secure mount feature?

Reply:
Yes, although if we're going to move it to the CDH, maybe we should do it that way from the start. I'll leave that for a kata reviewer to think about.

Reply:
Yeah, I have synced with @Xynnn007. He agreed to move the logic to CDH. I’ll handle it in this PR.

Reply:
Yes, we can reuse the script for multiple scenarios a) encrypted storage for extra image storage b) encrypted rbd for container volume. The both could get key a) from KBS or b) generate an ephemeral key.

Reply:
Ok, I think that's a good idea. It might take a little more work up front, but I think it will make things nicer down the road.

Reply:
https://github.com/confidential-containers/guest-components/issues/540